libsquoosh/src/index.js

```diff
@@ -8,6 +8,16 @@ import { autoOptimize } from './auto-optimizer.js';
 
 export { ImagePool, encoders, preprocessors };
 
+// The C++ code returns a string when an error occurs.
+// This function is used to evaluate return values from wasm,
+// and throw an error if applicable.
+function handleStringError(operation, value) {
```

Comment:
This seems like an asserting function. It should probably be called something like `assertNoCPPError` or something. `value` as first parameter, error message as second parameter, somewhat analogous to `console.assert()`. WDYT?

Reply:
Good point -- it really is an assertion function!